File: Lab_01/LAB01Q02 - Aquecimento Recursivo em Java/ContadorMaiusculasRecursivo.java
import java.util.Scanner;

public class ContadorMaiusculasRecursivo {

    public static int contarMaiusculasRecursivo(String str, int index) {
        if (index == str.length()) {
            return 0;
        }
        int count = Character.isUpperCase(str.charAt(index)) ? 1 : 0;
        return count + contarMaiusculasRecursivo(str, index + 1);
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        while (true) {
            String input = scanner.nextLine();
            if (input.equals("FIM")) {
                break;
            }
            int count = contarMaiusculasRecursivo(input, 0);
            System.out.println(count);
        }

        scanner.close();
    }
}

File: Lab_01/LAB01Q04 - Aquecimento Recursivo em C/ContadorMaiusculasRecursivo.c
#include <stdio.h>
#include <ctype.h>
#include <string.h>

int contarMaiusculasRecursivo(const char *str, int index) {
    if (index == strlen(str)) {
        return 0;
    }
    int count = isupper(str[index]) ? 1 : 0;
    return count + contarMaiusculasRecursivo(str, index + 1);
}

int main() {
    char input[100];

    while (1) {
        fgets(input, sizeof(input), stdin);
        input[strcspn(input, "\n")] = 0; // Remove newline character from input

        if (strcmp(input, "FIM") == 0) {
            break;
        }

        int count = contarMaiusculasRecursivo(input, 0);
        printf("%d\n", count);
    }

    return 0;
}
